### DISCUSSION TITLE
How to modify the nginx configuration

### DISCUSSION THREAD

**Discussion by la-jamesh** [jlesage/docker-firefox#185] [Q&A] [answered]
I'm seeing what code I need to add to modify the configuration file in the documentation but how to actually I implement that?

* Do I edit the nginx or default site config file after the container starts then reboot nginx? (When I try this it gives me an error about the map directive.)
* on build do I copy my own config file over to the `/opt/base/etc/nginx` conf?
* on build do I copy my own file to the `/etc/nginx/nginx/` dir?
* on build do I copy my own config file to the `/defaults/default_site.conf` when I tried this the container failed to boot

Any guidance for best practice on this would be appreciated. I would also be willing to open a PR to add to the current docs on how to follow that process.

**Comment by jlesage** [comment, marked as answer]
I'm not sure there is any "best practice", since the container is not designed to allow changes of the nginx configuration.

But if you want to hack something, you should do it to static files under `/opt/base/etc/nginx` and/or `/defaults/default*.conf`.
  * If you don't want to build the image yourself, you could map your own file at the creation of the container, via the `-v` parameter of the `docker run` command (e.g. `-v /home/user/default_site.conf:/defaults/default_site.conf`).
  * If you build the image yourself, you can provide your own copy of these files.

**Reply by la-jamesh** [reply]
Ahhh I see. Sorry I misunderstood the documentation. The documentation is saying that if you were to have your own reverse proxy using nginx this is how you configure it. Not "heres how you can change the config on the box"

Sorry for the silly question. I'll figure that out outside of the container then.
